rm-portal/src/test/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/CatalogRequestBuilderTest.java

```diff
@@ -0,0 +1,72 @@
+package org.ow2.proactive_grid_cloud_portal.rm.server.serialization;
+
+import org.apache.http.client.methods.HttpPost;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.ow2.proactive_grid_cloud_portal.rm.shared.CatalogConstants;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CatalogRequestBuilderTest {
+
+    @Mock
+    private CatalogObjectAction catalogObjectAction;
+
+    private File jsonFile;
+
+    private CatalogRequestBuilder catalogRequestBuilder;
+
+    @Before
+    public void setup() throws IOException {
+        this.catalogRequestBuilder = new CatalogRequestBuilder(catalogObjectAction);
+        this.jsonFile = Files.createTempFile("nodesource", ".json").toFile();
+        when(catalogObjectAction.getSessionId()).thenReturn("A4D5ca7vCAC582113WQCjmnW28");
+        when(catalogObjectAction.getNodeSourceName()).thenReturn("myNodeSource");
+        when(catalogObjectAction.getNodeSourceJsonFile()).thenReturn(this.jsonFile);
+        when(catalogObjectAction.getCommitMessage()).thenReturn("commit message");
+        when(catalogObjectAction.getKind()).thenReturn("NodeSource");
+        when(catalogObjectAction.getObjectContentType()).thenReturn("application/json");
+    }
+
+    @Test
+    public void testParametersAreSetForRevisedNodeSourceRequest() throws IOException {
+        when(this.catalogObjectAction.isRevised()).thenReturn(true);
+        HttpPost httpPost = this.catalogRequestBuilder.buildCatalogRequest("http://localhost:8080/catalog");
+        String string = getMultipartEntityString(httpPost);
+        checkThatRequestContainsRegularParameters(string);
+        assertThat("Request contains the " + CatalogConstants.KIND_PARAM + " parameter whereas it shouldn't", !string.contains(CatalogConstants.KIND_PARAM));
+        assertThat("Request contains the " + CatalogConstants.OBJECT_CONTENT_TYPE_PARAM + " parameter whereas it shouldn't", !string.contains(CatalogConstants.OBJECT_CONTENT_TYPE_PARAM));
```

Comment:
I am not sure, but maybe it possible to use predicate 'not()' instead of '!'

Reply:
Yes, it is. Great!

Reply:
But you must make sure to use `CoreMatchers.not`.